Src/Sarif.PatternMatcher/AnalyzeContext.cs

```diff
@@ -13,6 +13,16 @@ public class AnalyzeContext : AnalyzeContextBase
     {
         public static long FilesFilteredBySniffRegex;
 
+        /// <summary>
+        /// Gets or sets a dictionary linking file text with
+        /// A String8 that is used to in RE2 searching.
+        /// An array of bytes that comprise a buffer used in String8 conversion
```

Comment:
>array [](http://example.com/codeflow?start=15&length=5)

I think the intent was to encode some of this text as XML params args, but that wasn't done.

Reply:
I now understand that you simply moved this code. If you touched it, you bought it. :) The comment is bad.

Also, please do review this code standard with the team, i.e., property placement. Are we really going to enforce that standard? Why exactly? Someone on the team can advocate for it, I would like to hear the rationale.

Notice in this file as well that you moved a property with a code comment, but we're ignoring every other public thing that has no code comments. No good.

Now, balancing my remarks, if we get to a clean condition, we can enable warnings as errors and help prevent regression. So that is a fine value that might argue for not getting lost in too many 'you touched it' morasses.

Reply:
I do have to say the 'you touched it' idea can be a tad onerous and discourage doing a lot of cleanup that has high value but is too tedious to attack the owned content in the same pass.

I think there is great value in having multiple passes/waves of cleanup with different types of things done in each wave and if you mix too many things it makes it quite inefficient.

And personally, I would be very happy to NOT advocate for caring about code or property placement in a file.
It does however, feel messy to have fields/properties after the construction and methods. It does make sense to have private methods at the end of the class but that is a minor concern. There are so many subtle "this before this" rules it can become an almost unsolvable puzzle to find the perfect combination that satisfies all the placement rules.

Reply:
I have a preference for adding public methods before private ones: if someone else is reading the class to understand how to _interact_ with it (and not necessarily how it implements things internally), then having the public stuff at the top makes it easier to understand. One doesn't need to traverse different parts of the class. 

Personal preference here.